src/frontend/components/Felleskomponenter/MånedÅrVelger/MånedÅrVelger.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect } from 'react';
+
+import { MonthPicker, useMonthpicker } from '@navikt/ds-react';
+import { useSprakContext } from '@navikt/familie-sprakvelger';
+
+interface IProps {
+    tidligsteValgbareMåned?: Date;
+    senesteValgbareMåned?: Date;
+    label: React.ReactNode;
+    onChange: (date: Date) => void;
+}
+
+export const MånedÅrVelger: React.FC<IProps> = ({
+    tidligsteValgbareMåned,
+    senesteValgbareMåned,
+    label,
+    onChange,
+}) => {
+    const [valgtLocale] = useSprakContext();
+
+    const { monthpickerProps, inputProps, reset, selectedMonth } = useMonthpicker({
+        fromDate: tidligsteValgbareMåned,
+        toDate: senesteValgbareMåned,
+        locale: valgtLocale,
+        onMonthChange: (date?: Date) => date && onChange(date),
+    });
+
+    useEffect(() => {
+        if (selectedMonth) {
+            if (
+                (!!tidligsteValgbareMåned && tidligsteValgbareMåned > selectedMonth) ||
+                (!!senesteValgbareMåned && senesteValgbareMåned < selectedMonth)
+            ) {
+                reset();
+            }
+        }
+    }, [tidligsteValgbareMåned, senesteValgbareMåned]);
```

Comment:
Logikken for å resette et felt basert på et annet felt funker kun hvis man velger datoer ved å klikke i månedsvelgeren. Jeg får lov til å skrive inn en ugyldig dato manuelt, og det kan virke som det underliggende skjemafeltet som holder på verdien ikke blir satt, noe som gjør at feilmeldingen bruker får er litt rar 🙃

Lurer på om vi kan få en del gratis her hvis vi lager en implementasjon av Månedvelgeren som kombinerer sånn det er gjort for Månedvelger i ba-sak-frontend og for Datovelgeren her i ba-soknad. Syns eeegentlig det smudeste hadde vært å skrive om sånn at skjemafeltet ikke holdt på en ISOString, men heller et Date-objekt (sånn som vi har skrevet om i ba-sak og ks-sak) - men da må man i så fall også sørge for at det blir gjort om til isostring før man sender det til backend (og her også kanskje noe greier med mellomlagring, men er ikke så kjent med koblingen mellom skjemafeltene og det som mellomlagres)

Mye greier å skrive, så kanskje like greit å ta en muntlig prat i morgen i stedet 😆

![image](https://github.com/navikt/familie-ba-soknad/assets/25459913/76bcffbe-f146-45bd-b752-dee26dfa121f)